metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/zoho/ZohoAccessService.java

```diff
@@ -0,0 +1,230 @@
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import eu.europeana.corelib.definitions.edm.entity.Organization;
+import eu.europeana.corelib.solr.entity.Address;
+import eu.europeana.corelib.solr.entity.OrganizationImpl;
+import eu.europeana.enrichment.service.exception.ZohoAccessException;
+import eu.europeana.enrichment.service.zoho.model.ZohoOrganizationAdapter;
+import eu.europeana.metis.authentication.dao.PsqlMetisUserDao;
+import eu.europeana.metis.authentication.dao.ZohoAccessClientDao;
+import eu.europeana.metis.authentication.dao.ZohoFields;
+import eu.europeana.metis.exception.GenericMetisException;
+
+/**
+ * @author Simon Tzanakis (Simon.Tzanakis@europeana.eu)
+ * @since 2017-10-27
+ */
+@Service
+public class ZohoAccessService {
+
+	private static final String URL_ORGANIZATION_BASE = "http://data.europeana.eu/organization/base/";
+	private final ZohoAccessClientDao zohoAccessClientDao;
+
+	/**
+	 * Constructor of class with required parameters
+	 *
+	 * @param zohoAccessClientDao
+	 *            {@link ZohoAccessClientDao}
+	 * @param psqlMetisUserDao
+	 *            {@link PsqlMetisUserDao}
+	 */
+	@Autowired
+	public ZohoAccessService(ZohoAccessClientDao zohoAccessClientDao) {
+		this.zohoAccessClientDao = zohoAccessClientDao;
+	}
+
+	/**
+	 * This method retrieves OrganizationImpl object from Zoho organization
+	 * record.
+	 * 
+	 * @param jsonRecordResponse
+	 * @return representation of the Zoho organization record in OrganizatinImpl
+	 *         format
+	 * @throws IOException
+	 * @throws JsonMappingException
+	 * @throws JsonParseException
+	 */
+	protected Organization getOrganization(String organizationId) throws ZohoAccessException {
+
+		JsonNode jsonRecordsResponse;
+		try {
+			jsonRecordsResponse = zohoAccessClientDao.getOrganizationById(organizationId);
+		} catch (GenericMetisException e) {
+			throw new ZohoAccessException("Cannot get organization by id: " + organizationId, e);
+		}
+		JsonNode accountsNode = findRecordsByType(jsonRecordsResponse, ZohoFields.ACCOUNTS_MODULE_STRING);
+		JsonNode jsonRecord = accountsNode.findValue(ZohoFields.FIELDS_LABEL);
+
+		return getOrganizationFromJsonNode(jsonRecord);
+	}
+
+	/**
+	 * This method retrieves a list of organizations in {@link JsonNode} format.
+	 * 
+	 * @param jsonLeadsResponse
+	 * @param type
+	 *            the type of the objects to be retrieved
+	 * @return {@link JsonNode} representation of the organization
+	 */
+	JsonNode findRecordsByType(JsonNode jsonLeadsResponse, String type) {
+		if (jsonLeadsResponse.get(ZohoFields.RESPONSE_STRING).get(ZohoFields.RESULT_STRING) == null) {
+			return null;
+		}
+		return jsonLeadsResponse.get(ZohoFields.RESPONSE_STRING).get(ZohoFields.RESULT_STRING).get(type)
+				.get(ZohoFields.ROW_STRING);
+	}
+
+	Organization getOrganizationFromJsonNode(JsonNode jsonRecord) throws ZohoAccessException {
```

Comment:
This method looks something that an Adapter will do, like the ZohoOrganizationAdapter for example?

Reply:
It is actually copying the data from ZohoOrganizationAdapter (wrapper) into the corelib organization impl.  It also performs some data conversion through the called methods. So .. I would consider this to be business logic